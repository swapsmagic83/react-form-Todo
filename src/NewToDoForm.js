import React, {useState} from "react";

const NewToDoForm = ({addTask}) =>{

    const initial_state = {text:""}
    const [formData,setFormData] =useState(initial_state)
    const handleChange = (e) =>{
        const {name,value} = e.target
        setFormData(formData =>({
            ...formData,
            [name]:value
        }))
    }
    const handleSubmit =(e) =>{
        e.preventDefault()
        addTask({...formData})
        setFormData(initial_state)
    }

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="text">Task</label>
            <input
            id="text"
            type="text"
            name="text"
            value={formData.text}
            placeholder="Enter task here"
            onChange={handleChange}></input>
            <button>Add</button>
        </form>
    )
}

export default NewToDoForm