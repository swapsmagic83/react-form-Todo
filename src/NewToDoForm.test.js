import React from "react";
import { render } from "@testing-library/react";
import NewToDoForm from "./NewToDoForm";

it("renders without crashing", function(){
    render(<NewToDoForm />)
})
it("matches snapshot",function(){
    const { asFragment } = render(<NewToDoForm />)
    expect(asFragment()).toMatchSnapshot()
})