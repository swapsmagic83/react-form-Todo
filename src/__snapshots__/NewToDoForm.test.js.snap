// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <button>
      Remove
    </button>
  </div>
</DocumentFragment>
`;
